packages/envconfig/src/envconfig-toml.ts

```diff
@@ -0,0 +1,326 @@
+import { readFileSync } from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import { parse, stringify, TomlTable } from 'smol-toml';
+import {
+  ConfigDataSource,
+  LoadClientConfigOptions,
+  LoadClientProfileOptions,
+  TomlClientConfig,
+  TomlClientConfigCodec,
+  TomlClientConfigProfile,
+  TomlClientConfigTLS,
+} from './types';
+import { normalizeGrpcMetaKey } from './utils';
+
+function sourceToStringData(source?: ConfigDataSource): string | undefined {
+  if (source === undefined) {
+    return undefined;
+  }
+  if ('path' in source) {
+    return readFileSync(source.path)?.toString();
+  }
+
+  if (Buffer.isBuffer(source.data)) {
+    return source.data.toString();
+  }
+  return source.data;
+}
+
+type EnvProvider = { kind: 'Map'; map: Record<string, string> } | { kind: 'System' };
+
+function getEnvVar(provider: EnvProvider, key: string): string | undefined {
+  if (provider.kind === 'Map') {
+    return provider.map[key];
+  }
+  return process.env[key];
+}
+
+export function tomlLoadClientConfig(options: LoadClientConfigOptions): TomlClientConfig {
+  const envProvider: EnvProvider = options.overrideEnvVars
```

Comment:
Why not:

```
const env = options.overrideEnvVars ?? process.env;
```

And then you can simply do `env[key]` everywhere; no need for the `EnvProvider` type, nor the `getEnvVar` function.